Lib/ast.py

```diff
@@ -1053,15 +1054,82 @@ def visit_AsyncWith(self, node):
         with self.block(extra=self.get_type_comment(node)):
             self.traverse(node.body)
 
+    def _str_literal_helper(
+        self, string, quote_types=("'", '"', '"""', "'''"), escape=""
+    ):
+        """Helper for writing string literals, minimizing escapes.
+        Returns (possible quote types, string literal to write).
+        """
+        # Escape characters we've been told to escape, backslashes, and
+        # non-printable characters other than \n and \t.
+        escape = {*escape, '\\'}
+        escaped_string = "".join(
+            (
+                c.encode('unicode_escape').decode('ascii')
+                if c in escape or (not c.isprintable() and c not in '\n\t')
+                else c
+            )
+            for c in string
+        )
```

Comment:
As I said in the previous review, I still find this part very hard to read. Like `esc_char` you removed below, I think it would be best if we can make it a function with 'pure' conditions.
```diff
+        def escape_char(char):
+            if not escape_special and char in "\n\t":
+                return char
+            elif char == "\\" or not char.isprintable():
+                return char.encode("unicode_escape").decode("ascii")
+            else:
+                return char
+        escaped_string = "".join(map(escape_char, string))
```